site/public/js/calendar.js

```diff
@@ -104,6 +109,67 @@ function generateCalendarItem(item) {
         </a>`;
 }
 
+function generateItemHtml(item, date) {
+    let styleString = '';
+    switch (item['type']) {
+        case 'announcement':
+            styleString = 'style="font-weight: bold;"';
+            break;
+        case 'important':
+            styleString = 'style="font-style: italic;"';
+            break;
+    }
+    if (isGlobal) {
+        return `<p ${styleString}>${item['course']}: ${item['text']}`;
+    }
+    else {
+        if (isInstructor) {
+            const string =  `<p ${styleString}>${item['text']} <a data-id="${item['id']}" data-type="${item['type']}" data-text="${window.btoa(item['text'])}" data-date="${date}" class="calendar-item-edit-button"><i class="fas fa-pencil-alt"></i></a></p>`;
+            /*document.getElementById(item['id']).onclick = function () {
+                editCalendarItemForm(item['type'], item['text'], item['id'], date);
+            };*/
+            return string;
+        }
+        else {
+            return `<p ${styleString}>${item['text']}</p>`;
+        }
+    }
+}
+
+function editCalendarItemForm(itemType, itemText, itemId, date) {
+    $('#calendar-item-type-edit').val(itemType);
+    $('#calendar-item-text-edit').val(itemText);
+    $('#edit-picker-edit').val(date);
+    $('#calendar-item-id').val(itemId);
+
+    $('#edit-calendar-item-form').css('display', 'block');
```

Comment:
It would be equivalent to use jQuery's `.toggle()` method here, which also avoids accidentally overriding someone's CSS in the future if the display property was set to something like `inline-block`.  Minor change though so if it's not easy to implement elsewhere, don't worry about it.

Reply:
I'm going to use .show() here instead as there is no reason this button should hide for whatever reason.